### ISSUE TITLE
Is there a way of updating the generated Swagger spec?

### ISSUE THREAD

**Issue by Albert-Gao** [tiangolo/fastapi#5551]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()
app.openapi_version = "3.1.0"
```

### Description

Hi, we have an internal tooling of linting the generated swagger (aka, OpenAPI) doc, which requires the API spec to follow `3.1.0`. However, the current generated version is [`3.0.2`](https://github.com/tiangolo/fastapi/blob/f67b19f0f73ebdca01775b8c7e531e51b9cecfae/fastapi/applications.py#L106).

I used the above code to update it to `3.1.0`, and it works. 😄

However, it looks like I am hard coding the version number, rather than generating with the real `v3.1.0` spec, so I wonder is there a way to update the internal swagger spec version?

Thanks 😄

### Operating System

Windows

### FastAPI Version

0.65.2

### Python Version

3.8.8

**Comment by yinziyan1206** [comment]
the swagger is embedded in FastAPI server. if you want to change it, overwrite the app class or override the method

**Comment by tiangolo** [comment]
Hey there! Yep, differences are but too big in the specs. And in fact, FastAPI does a couple of things a tiny bit more compatible with 3.1.0.

The only reason why I haven't updated it is because Swagger UI breaks. But once they support it I'll be able to update the generated version.

For now, you should be mostly fine by simply overriding it. I know it feels weird, but it shouldn't be too problematic if at all.

**Comment by Albert-Gao** [comment]
Thanks for the asnwers :) 
Turns out the hack does not work. Received an error message, the swagger is forcing the version range to be below `3.1.0`, thanks for the explanation. We've delayed the ticket. Thanks for this framework, simply the best web framework i've ever used across js/golang/python! love it!

**Comment by iudeen** [comment]
@Albert-Gao please close the issue if you are happy with the answer :)

**Comment by Albert-Gao** [comment]
Sure :) @iudeen 
Just 1 more question.
is this the PR I should track this issue?
https://github.com/tiangolo/fastapi/pull/3038

**Comment by iudeen** [comment]
Yes, that PR aims to add support, but is on hold till Swagger UI is unblocked (given in PR's description)!

**Comment by Albert-Gao** [comment]
Thanks, closed :)

**Comment by tiangolo** [comment]
Thanks! I'm very glad you have liked it like that! 😁😊 And thanks for coming back to close the issue.

Also thanks @iudeen for the help! 🍰
